lib/puppet/indirector/yaml.rb

```diff
@@ -19,17 +19,20 @@ def find(request)
   def save(request)
     raise ArgumentError.new("You can only save objects that respond to :name") unless request.instance.respond_to?(:name)
 
-    file = path(request.key)
+    files = path(request.key)
```

Comment:
My testing shows that those operations are never called on a Indirector::Report::Yaml object, that is to say the lastrunreport is only ever involved in `#save` calls.